src/init.rs

```diff
@@ -1,28 +1,14 @@
 extern crate ffmpeg_next as ffmpeg;
 
 use ffmpeg::{
-  format::{
-    version,
-    register_all,
-  },
   error::register_all as register_all_errors,
 };
 
 use super::ffi::init_logging;
 
-/// The version of `libavformat` that deprecated `register_all`.
-const AVFORMAT_MAX_VERSION_REQUIRES_REGISTER_ALL: u32 = 3803492;
-
 /// Initialize global ffmpeg settings. This also intializes the
 /// logging capability and redirect it to `tracing`.
-/// 
-/// In older versions, this will invoke `register_all`.
 pub fn init() {
-  // Older versions of ffmpeg require this.
-  if version() < AVFORMAT_MAX_VERSION_REQUIRES_REGISTER_ALL {
-    register_all();
```

Comment:
Is it possible to still have this check when compiling for ffmpeg4 so we retain compatibility with older ffmpeg versions?

Reply:
In 5.0 it was removed, but kept around under a compile flag for backwards compatibility

A compile flag could similarly be added here, but that would require all <5.0 or 5.0 users to specify it. What you think?

Reply:
Scratch that, apparently ffmpeg calls `register_all` automatically when it detects that ffmpeg is <5.0 on init. So this is safe to remove

https://github.com/zmwangx/rust-ffmpeg/blob/8bb9f7c7ee649a024f701aebddc79878fae4a4b5/src/lib.rs#L109-L118

Reply:
Added ffmpeg 3.4 to test pipeline just to be sure though